piff/basis_interp.py

```diff
@@ -414,22 +438,44 @@ class BasisPolynomial(BasisInterp):
-                        squares solution.  QR decomposition requires more memory than the default
-                        and is somewhat slower (nearly a factor of 2); however, it is significantly
-    :param use_jax:     Use JAX for solving the linear algebra equations rather than numpy/scipy.
-                        Therefore, it may be preferred for some use cases. [default: False]
-    :param logger:      A logger object for logging debug info. [default: None]
+    :param order:           The order to use for each key.  Can be a single value (applied to all
+                            keys) or an array matching number of keys.
+    :param keys:            List of keys for properties that will be used as the polynomial arguments.
+                            [default: ('u','v')]
+    :param max_order:       The maximum total order to use for cross terms between keys.
+                            [default: None, which uses the maximum value of any individual key's order]
+    :param solver:          Which solver to use to solve linear algebra of interpolation. Solvers
+                            available are "scipy", "qr", "jax", "cpp", "cpp32".
+                            "scipy": Default. Use scipy to solve.
+                            "qr": Use QR decomposition for the solution rather than the more direct least
+                            squares solution.  QR decomposition requires more memory than the default
+                            and is somewhat slower (nearly a factor of 2); however, it is significantly
+                            less susceptible to numerical errors from high condition matrices.
+                            "jax": Use JAX for solving the linear algebra equations rather than numpy/scipy.
+                            Equivalent to "scipy" solver therefore, it may be preferred for some use cases. It will
+                            be faster than "scipy" solver on multi-core cpu / gpu are available.
+                            "cpp": Use C++/eigen for solving the linear algebra equations rather than
+                            numpy/scipy Equivalent to "scipy" solver, therefore, it may be preferred for some
+                            use cases. On a single core cpu (and more), it will be faster than "scipy" solver
+                            if the number of training stars is more than ~30. With a Piff config using `PixelGrid`
+                            with `size=25` and `interp="Lanczos(11)"` and a second order polynomial for interpolation,
+                            and running on ~O(100) PSFs reserved stars on a 4GB single core CPU, "cpp" solver is 60%
+                            faster than "scipy" solver.
+                            "cpp32": Same as cpp solver but use float32, faster than cpp with a trade for accuracy.
+                            Great power imply great responsability, to use carefully.
```

Comment:
Most of this discussion should be moved up to the text section above the parameters list.  So this can be a short doc that ends at "Solvers available are "scipy", "qr", "jax", "cpp", "cpp32".  See above for details."